earthly/rust/stdcfgs/clippy.toml

```diff
@@ -1,4 +1,5 @@
 allow-unwrap-in-tests = true
 allow-expect-in-tests = true
 allow-panic-in-tests = true
-arithmetic-side-effects-allowed = ["num_bigint::BigInt"]
+allow-indexing-slicing-in-tests = true
+arithmetic-side-effects-allowed = ["num_bigint::BigInt", "cardano_blockchain_types::Slot"]
```

Comment:
Is it possible to just disable the lint on the code with an annotation, rather than add the exception here.  I would rather we only use this list for code we don't control.

Reply:
Yes, it is possible to use annotations and it is what we currently do, but it is also possible to do that for `BigInt`. Personally I think that it is annoying to put annotations in multiple places even if we know that it is safe. Currently we have multiple annotations like this one:
```rust
// It is ok because slot implement saturating subtraction.
#[allow(clippy::arithmetic_side_effects)]
let slot_rollback_size = head_slot - rollback_slot;
```

Reply:
Anyway, I have removed `Slot` from this list. 🙃

Reply:
Putting it in the list means there is no way to know what code is exactly causing the issue, its globally disabling it.  Putting the annotation means this EXACT line is causing this issue and allows it to be properly validated.

In the above example thats an indication the API isn't great for that type.
instead it might be better:
```
let slot_rollback_size = head_slot.rollback_size(rollback_slot);
```
no need for the annotation.

Not suggesting this change be made, just that there are different ways to approach this.  I would much rather we be explicit so we can properly validate that the annotation is in fact safe and doing what we would expect, rather than silently squashing all instances of it and hoping one of those is not a problem.